pkg/plugins/runtime/gateway/filter_chain_generator.go

```diff
@@ -104,31 +122,55 @@ func (g *HTTPSFilterChainGenerator) Generate(
 					return nil, nil, errors.Errorf("duplicate TLS certificate %q", secret.Name)
 				}
 
-				hostResources.Add(NewResource(secret.Name, secret))
+				resource := NewResource(secret.Name, secret)
+
+				hostResources.Add(resource)
+
+				downstream.CommonTlsContext.TlsCertificateSdsSecretConfigs = append(
+					downstream.CommonTlsContext.TlsCertificateSdsSecretConfigs, envoy_tls_v3.NewSecretConfigSource(resource.Name),
+				)
 			}
 
 		case mesh_proto.MeshGateway_TLS_PASSTHROUGH:
+			builder.Configure(
+				envoy_listeners.MatchServerNames(host.Hostname),
+			)
+
 			// TODO(jpeach) add support for PASSTHROUGH mode.
 			return nil, nil, errors.Errorf("unsupported TLS mode %q", host.TLS.GetMode())
 
-		default:
-			return nil, nil, errors.Errorf("unsupported TLS mode %q", host.TLS.GetMode())
-		}
+		case mesh_proto.MeshGateway_TLS_NONE:
+			if !info.Listener.CrossMesh {
```

Comment:
so is it expected that cross mesh is set by putting TLS to None and crossMesh=true? I thought it will be TLS terminated + crossMesh=true

Reply:
There was a convo in the proposal about this. For cross mesh, that the listener uses TLS is an implementation detail, the same way that every Datapläne uses it but is treated as HTTP. Do you think this is consistent? Validation would prevent anything else for now.

I suppose one could imagine using TLS with a custom cert for a crossMesh listener, in which case you'd need to set TLS terminated mode with crossMesh

Reply:
added validation for this

Reply:
It's now no longer dependent on the gateway/listener tags